From b470ecceb52b526260b0e34ee0017332a77b283c Mon Sep 17 00:00:00 2001
From: Qiang Yu <Qiang.Yu@amd.com>
Date: Thu, 10 Aug 2017 15:13:07 +0800
Subject: [PATCH 43/46] Revert "amdgpu:fix potential deadlock"

This reverts commit eaea24b7acfb8d2cb1db8baa8f353d9fcc69a2b6.

The reverted commit is submitted by mistaken.

Signed-off-by: Qiang Yu <Qiang.Yu@amd.com>
---
 amdgpu/amdgpu_bo.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/amdgpu/amdgpu_bo.c b/amdgpu/amdgpu_bo.c
index c973a0d..09028c7 100644
--- a/amdgpu/amdgpu_bo.c
+++ b/amdgpu/amdgpu_bo.c
@@ -673,10 +673,10 @@ int amdgpu_bo_cpu_map(amdgpu_bo_handle bo, void **cpu)
 		pthread_mutex_unlock(&bo->cpu_access_mutex);
 		return -errno;
 	}
+	amdgpu_add_handle_to_table(bo);
 	bo->cpu_ptr = ptr;
 	bo->cpu_map_count = 1;
 	pthread_mutex_unlock(&bo->cpu_access_mutex);
-	amdgpu_add_handle_to_table(bo);
 
 	*cpu = ptr;
 	return 0;
-- 
2.7.4

